schema/proto_test.go

```diff
@@ -0,0 +1,128 @@
+package schema
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/xitongsys/parquet-go/common"
+	"github.com/xitongsys/parquet-go/parquet"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type JobStatus int32
+
+const (
+	JobStatus_JobStatus_UNSPECIFIED  JobStatus = 0
+	JobStatus_BLOCKED                JobStatus = 1
+	JobStatus_ENQUEUED               JobStatus = 2
+	JobStatus_RUNNING                JobStatus = 3
+	JobStatus_COMPLETED              JobStatus = 4
+	JobStatus_ERRORED                JobStatus = 5
+	JobStatus_CANCELLED              JobStatus = 6
+	JobStatus_UPSTREAM_NOT_PROCESSED JobStatus = 7
+)
+
+func (x JobStatus) Enum() *JobStatus {
+	p := new(JobStatus)
+	*p = x
+	return p
+}
+
+type ClassNoTag struct {
+	Name   string
+	Number int
+	Score  string
+}
+
+type StudentNoTag struct {
+	Name    string
+	Age     int
+	Classes []*ClassNoTag
+	Info    *map[string]*string
+	Sex     *bool
+}
+
+type ProtoMessage struct {
+	Timestamp timestamppb.Timestamp
+	Status    JobStatus
+	IntVal    int32
+}
+
+func TestProtoSpecificSchema(t *testing.T) {
+	schemaHandler, err := NewSchemaHandlerFromProtoStruct(new(ProtoMessage))
+	if err != nil {
+		t.Errorf("failed to generate schema handler: %v", err)
+	}
+	assert.Equal(t, 3, len(schemaHandler.ValueColumns))
+	assert.Equal(t, parquet.Type_INT64, *schemaHandler.SchemaElements[1].Type)
+	assert.Equal(t, parquet.ConvertedType_TIMESTAMP_MILLIS, *schemaHandler.SchemaElements[1].ConvertedType)
+	assert.Equal(t, parquet.Type_BYTE_ARRAY, *schemaHandler.SchemaElements[2].Type)
```

Comment:
This is because given an Enum, we'll store the **string value** in Parquet, right? e.g. `JobStatus_ENQUEUED` as a string in the Parquet record.